### PR TITLE
#943 Started Implementing BitBucket.

### PR DESCRIPTION
**Pull request by criske**
Left puzzles for Repo, Invitations and Organizations implementations.

Pr for #943

### CODE REVIEW

**Comment by charlesmike** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2021-02-09T11:13:01.719369``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: self-core-impl/src/main/java/com/selfxdsd/core/AccessToken.java [hunk 1]
```diff
@@ -83,4 +83,35 @@ public String value() {
...
+         * Header value.
+         */
+        private final String value;
+
+        /**
+         * Ctor.
+         *
+         * @param value Header value.
+         */
+        public Bitbucket(final String value) {
```
Comment (criske) [inline self-core-impl/src/main/java/com/selfxdsd/core/AccessToken.java:103]: @amihaiemil Unlike gh/gl, bitbucket tokens expire in 1h https://developer.atlassian.com/bitbucket/api/2/reference/meta/authentication#refresh-tokens.  We may need to pass the 'value' through a `Supplier` that will ensure the token is always up to date.

Comment (amihaiemil) [reply self-core-impl/src/main/java/com/selfxdsd/core/AccessToken.java:103]: @criske good point, I'll keep it noted.

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**